modules/custom/activity_viewer/src/Plugin/views/filter/ActivityNotificationVisibilityAccess.php

```diff
@@ -183,17 +183,58 @@ public function query() {
       $or->condition($comments_on_content);
     }
 
+    // For likes we can safely assume these end up only in the recipient user,
+    // which is usually the entity owner which got liked.
     if ($account->isAuthenticated()) {
       $vote_access = new Condition('AND');
       $vote_access->condition('activity__field_activity_entity.field_activity_entity_target_type', 'vote');
-      $vote_access->condition('activity__field_activity_recipient_user.field_activity_recipient_user_target_id', $account->id());
-
+      $vote_access->condition('activity__field_activity_recipient_user.field_activity_recipient_user_target_id', (string) $account->id());
       $or->condition($vote_access);
     }
 
+    // For an enrollment we match the recipient uid for enrollments
+    // activities are only created for those with the OrganizerActivityContext
+    // see getRecipientOrganizerFromEntity.
+    if ($account->isAuthenticated()) {
```

Comment:
Just a small suggestion: We should add `$authenticated = $account->isAuthenticated();`, this will avoid multiple function calls at different lines.